server/py/framework/db/sqldb/db.py

```diff
@@ -6943,12 +6937,14 @@ def _is_table_exists(session: Session, table_name: str) -> bool:
 
     @staticmethod
     def _paginate_query(
-        query, page: typing.Optional[int] = None, page_size: typing.Optional[int] = None
+        query, offset: typing.Optional[int] = None, limit: typing.Optional[int] = None
     ):
-        if page is not None:
-            page_size = page_size or config.httpdb.pagination.default_page_size
-            if query.count() < page_size * (page - 1):
+        if offset:
+            if query.count() < offset:
                 raise StopIteration
```

Comment:
As discussed, while not related to this change - this generates another query to the DB which is resource intensive and should be avoided. It seems this doesn't really generate value, so should be removed.